src/web_interface/file_tree/file_tree.py

```diff
@@ -148,12 +148,23 @@ class VirtualPathFileTree:
 
     def __init__(self, root_uid: str, parent_uid: str, fo_data: FileTreeData, whitelist: list[str] | None = None):
         self.uid = fo_data.uid
-        self.root_uid = root_uid if root_uid else list(fo_data.virtual_file_path)[0]
+        self.root_uid = root_uid if root_uid else self._find_root_uid(parent_uid, fo_data)
         self.parent_uid = parent_uid
         self.fo_data: FileTreeData = fo_data
         self.whitelist = whitelist
         self.virtual_file_paths = self._get_virtual_file_paths()
 
+    @staticmethod
+    def _find_root_uid(parent_uid: str, fo_data: FileTreeData) -> str:
+        '''
+        If we don't have a rood_uid, we must find a root_uid that contains the parent_uid (we can't just take a
+        random one because then the files could be missing from the file tree).
+        '''
+        for root_uid, vfp_list in fo_data.virtual_file_path.items():
+            if any(parent_uid in vfp for vfp in vfp_list):
+                return root_uid
+        return list(fo_data.virtual_file_path)[0]  # safety fallback: this should not occur under normal circumstances
```

Comment:
If it "should not occur under normal circumstances" can't we just throw an exception?

What are unnormal circumstances where this could happen?

Reply:
I don't think we should let it crash at this point but maybe logging an error or warning would be appropriate (but this could also be a new source for spam in the logs so I'm hesitant). Also I'm not sure when this could happen. Maybe if you look at a firmware while deleting it or something like that :sweat_smile:

Reply:
If you look at the requirements that I described that are needed for this to happen, it was a rather rare case already.